constrained-voucher.mkd

```diff
@@ -520,6 +506,92 @@ One key difference is that the names of the leaves in the YANG does not have a m
+Validation of this is what takes the connection out of the provisional state (see {{I-D.ietf-anima-bootstrapping-keyinfra}} section 5.6.2).
+
+The voucher needs to be validated first.
+The Pledge needs to have a public key to validate the signature from the MASA on the voucher.
+
+Certificate chains may be included in the COSE-signed Constrained Vouchers described in this document, using the x5bag attribute from {{I-D.ietf-cose-x509}}.
+However, this would require that the Pledge is able to form a certificate chain and validate it against a built-in Trust Anchor.
+
+A simpler method is that the MASA would sign the Voucher directly with a key pair for which the Pledge has the public key built-in to its firmware.
+That is, the MASA would use a degenerate PKI of level one.
+
+This has significant advantages on the Pledge, but presents security issues for the MASA, as it needs to keep a private online for signing purposes.
+Loss or compromise of that key would result in significant damage to the BRSKI mechanism.
+
+There are a few mitigations that can be applied.
+
+The first is that different Pledges can be loaded with different anchors.
+For instance, if there are ten key pairs loaded into the MASA, numbered 0 through 9, then Pledges with serial numbers ending in the same digit would use only the corresponding key.
+This can be generalized to large numbers, with each addition reducing the number of Pledges affected by a single compromise.
+
+Keys can also be rotated over time, with new keys generated each week, or according to some other schedule known only to the manufacturer.
+The exact mapping can remain confidential to the manufacturer.
+
+The above mechanism would not help with loss or compromise if every key was loaded in the MASA.
+As the IDevID contains a pointer to the MASA, each Pledge can also refer the Registrar to a specific MASA (the one with the right key loaded), by changing the MASA URL encoded into the IDevID.
+This is a form of sharding: it may also have benefits when it comes to managing load when it is expected that many thousand of devices per second will attempt to onboard.
+
+The document {{I-D.richardson-anima-masa-considerations}} section 2.4 has more details.
+
+## Enrollment
+
+This document makes the use of the /crts call to retrieve an CSR Attributes object optional.
+
+A Pledge that does not do this does not need to change the Certificate Signing Request (CSR) that it sends to the Registrar.
+The ASN.1 bits of the CSR can therefore be treated almost as a binary blob, signed with the IDevID.
+
+(EDNOTE: Can the signing even be done ahead of time?  Is there any live proof of possession?)
+
+The resulting LDevID that is received, if arranged to be paired with the existing private key used for the IDevID, can then be treated as a binary blob to be sent when authenticating other
```

Comment:
See my comment above; the "paired" part here is unclear to me since the keypair used for the LDevID differs from the IDevID keypair.

Reply:
See above comments on "binary blob", can work but only in those specific cases.